core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -447,6 +448,21 @@ public TaskManagerConfig setScaleWritersMaxWriterCount(int scaleWritersMaxWriter
         return this;
     }
 
+    @Min(1)
+    public int getScaleWritersMaxNodesCount()
+    {
+        return scaleWritersMaxNodesCount;
+    }
+
+    @Config("task.scale-writers.max-nodes-count")
+    @ConfigDescription("Maximum number of nodes that will take part in writer tasks. It is an upper bound on scaling of writers " +
+            "and works only if task.scale-writers.enabled is set")
```

Comment:
Node limit should also apply `task.scale-writers.enabled` when `preferred partitioning` insert mode is used. Please refer to @gaurav8297 for help